Mathlib/Data/Set/Function.lean

```diff
@@ -728,6 +735,22 @@ lemma graphOn_insert (f : α → β) (x : α) (s : Set α) :
 lemma image_fst_graphOn (f : α → β) (s : Set α) : Prod.fst '' graphOn f s = s := by
   simp [graphOn, image_image]
 
+@[simp] lemma image_snd_graphOn (f : α → β) : Prod.snd '' s.graphOn f = f '' s := by ext x; simp
+
+lemma fst_injOn_graph : (s.graphOn f).InjOn Prod.fst := by aesop (add simp InjOn)
+
+lemma graphOn_comp (s : Set α) (f : α → β) (g : β → γ) :
+    s.graphOn (g ∘ f) = (fun x ↦ (x.1, g x.2)) '' s.graphOn f := by
+  simpa using image_comp (fun x ↦ (x.1, g x.2)) (fun x ↦ (x, f x)) _
+
+lemma graphOn_univ_eq_range : univ.graphOn f = range fun x ↦ (x, f x) := image_univ
+
+lemma graphOn_univ_injective : Injective (univ.graphOn : (α → β) → Set (α × β)) := by
+  aesop (add simp [Injective, Set.ext_iff])
+
+@[simp] lemma graphOn_univ_inj {g : α → β} :
```

Comment:
This suggests there should be a (simp) lemma saying `s.graphOn f = s.graphOn g ↔ s.EqOn f g`, then this would follow easily from that. What do you think?